python/src/main/java/org/apache/apex/malhar/python/base/ApexPythonEngine.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.python.base;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+
+import org.apache.apex.malhar.python.base.requestresponse.PythonInterpreterRequest;
+import org.apache.apex.malhar.python.base.requestresponse.PythonRequestResponse;
+
+/**
+ * Defines the methods that needs to be implemented by the Python operator implementations. The first implementation
+ *  takes the approach of in-memory interpreter using JEP. Other possibilities are using Py4J which is an inter process
+ *   communication model.
+ */
+public interface ApexPythonEngine
+{
+  /***
+   * Used to perform any pre interpreter processing.
+   * @param preInitConfigs The configuration that is going to be used by the interpreter
+   * @throws ApexPythonInterpreterException if there is an issue in executing the pre interpreter logic
+   */
+  void preInitInterpreter(Map<PythonInterpreterConfig,Object> preInitConfigs) throws ApexPythonInterpreterException;
+
+  /***
+   * Starts the interpreter.
+   * @throws ApexPythonInterpreterException if library not locatable or any other issue starting the interpreter
+   */
+  void startInterpreter() throws ApexPythonInterpreterException;
+
+  /***
+   * Used to perform any logic that needs to be executed after the interpreter is started but before any tuples start
+   *  getting processed. Example, setting the starting state of the variables that are used in tuple processing.
+   * @throws ApexPythonInterpreterException
+   */
+  void postStartInterpreter() throws ApexPythonInterpreterException;
+
+  /***
+   * Runs a series of commands.
+   * @param executionMode Whether these commands need to be run on all worker nodes or any of the worker node
+   * @param windowId used to select the worker from the worker pool.Can be any long if an operator is not using this.
```

Comment:
windowId and requestId need more explanation. Not clear how they "select" a worker?

Reply:
Added more javadocs to explain the usage of windowId and requestId. The implementation needed a way to determine the best fit worker basing on the worker execution mode. 1. BROADCAST/ANY/STICKY.

If it is broadcast, then these values are not used at all.
If it is any worker, the windowid and requestid are used to roundrobin requests.
if it is sticky, then the combination of these values is used to select a worker.